util/base/src/main/java/jetbrains/jetpad/base/Async.java

```diff
@@ -15,11 +15,33 @@
  */
 package jetbrains.jetpad.base;
 
+import com.google.common.base.Function;
+
 /**
  * Asynchronous computation
  */
 public interface Async<ItemT> {
   Registration onSuccess(Handler<? super ItemT> successHandler);
   Registration onResult(Handler<? super ItemT> successHandler, Handler<Throwable> failureHandler);
   Registration onFailure(Handler<Throwable> failureHandler);
+
+  /**
+   * This method must always create new async every time it's called.
+   *
+   * @param success   handler to transform async result.
```

Comment:
I feel that "@param" section here and in then() method are redundant.
What do you think about adding a comment about catching possible exceptions during "success" functions calls?